### PR TITLE
L036 refinement - FROM clause interaction

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

This PR aims to resolve an issue with the logic in L036, in which the columns in a query such as `SELECT a, b FROM c;` are moved to a new line without considering the impact on the FROM clause, leaving the FROM clause hanging off to the right of the final column rather than under the SELECT keyword as expected.

This is an example of behaviour prior to this PR:
Screen-Recording-2021-11-13-at-02 02 39

This is an example of the updated logic to account for moved FROM clauses:
Screen-Recording-2021-11-13-at-02 05 17

I have included the additional test cases to validate the new correct behaviour.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by jpy-git** [comment]
@barrywhart Could you take a look at this when you get the chance, thanks 😄

File: src/sqlfluff/rules/L036.py [hunk 1]
```diff
@@ -82,12 +90,21 @@ def _get_indexes(segment):
             ):
                 first_whitespace_idx = fname_idx
 
+        for seg in context.siblings_post:
+            if seg.is_type("from_clause"):
```
Comment (barrywhart) [inline src/sqlfluff/rules/L036.py:94]: Nitpick: Would it be clearer to change this to `if seg is from_segment:` ?

File: src/sqlfluff/rules/L036.py [hunk 2]
```diff
@@ -333,5 +369,33 @@ def _eval_single_select_target_element(
...
+                and (
+                    select_clause.pos_marker.working_line_no
+                    != select_targets_info.from_segment.pos_marker.working_line_no
+                )
+                and (
+                    wildcard_select_clause_element.pos_marker.working_line_no
+                    == select_targets_info.from_segment.pos_marker.working_line_no
+                )
+            ):
+                fixes = [
```
Comment (barrywhart) [inline src/sqlfluff/rules/L036.py:390]: Is this the same code as line 147? I wonder if we ought to create a function with this code and call it from both these places.

**Review by barrywhart — APPROVED** [review]
A couple of very minor suggestions, but I'm happy to merge "as is" -- let me know what you think.

**Comment by barrywhart** [comment]
BTW, nice work!!

File: src/sqlfluff/rules/L036.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/rules/L036.py]: I've implemented this change 👍

File: src/sqlfluff/rules/L036.py (hunk 2 above)
Comment (jpy-git) [reply src/sqlfluff/rules/L036.py:390]: So my thinking on this was:
On lines 142-157, we are appending these lint fixes to an already existing lint fixes, whereas on lines 378-399 this condition only a wildcard and creates a separate list of fixes to the rest of the code. Given that the conditions in the if clause are also different, the only thing we could add to a function would be the
```python
 fixes.extend(
                        [
                            LintFix("delete", ws)
                            for ws in select_targets_info.pre_from_whitespace
                        ]
                    )
                    fixes.append(
                        LintFix(
                            "create", select_targets_info.from_segment, NewlineSegment()
                        )
                    )
```
lines, which IMO are already pretty self explanatory so maybe not much is gained in terms of moving them to a separate method?

Happy to implement the change if you think it's worth it though 😄

**Comment by jpy-git** [comment]
@barrywhart Thanks, appreciate the feedback! 😄 I think the PR is ready to go if you're happy with my reasoning on the splitting fix logic into new method point?

**Comment by barrywhart** [comment]
Looks good. No word worries about the new function -- truly nitpicky feedback on my part. I'll do a final look and merge tonight.

**Pull request merged**